hierarchies.rst

```diff
@@ -0,0 +1,409 @@
+.. _hierarchies:
+
+=================================
+Working with Flux Job Hierarchies
+=================================
+
+A key feature of Flux is the ability to run an instance to manage a workload
+in almost any environment. From a session started on a laptop with ``flux
+start --test-size=1``, to a batch job running across thousands of nodes
+on a cluster, Flux provides a consistent interface to resources.
+
+These resources and the work assigned to them may then be further
+divided by running new instances of Flux as jobs via ``flux mini batch``,
+``flux mini alloc``, or ``flux mini submit ... flux start``, each of which
+can further divide resources through more instances, and so on. Since an
+instance of Flux has its own scheduler and configuration, this can aid in
+job throughput or scheduler specialization when complex systems of jobs
+are being handled.
+
+However, even for a small hierarchy of jobs, understanding the state of
+running instances can be a challenge, so Flux offers several high level
+concepts and tools to ease their management.
+
+This guide gives an introduction to managing Flux job hierarchies, whether
+they are initiated under a Flux system instance, a foreign resource manager,
+or under a test session running on a laptop.
+
+----------------------
+Instance level (depth)
+----------------------
+
+A Flux instance records its depth in the current job hierarchy in the
+``instance-level`` attribute. This attribute can be useful when descending
+a hierarchy of jobs. In this guide, the following value of ``PS1`` will
+be used to indicate the size and depth of the current enclosing instance
+for reference purposes:
```

Comment:
something like "the initial instance, such as the system instance or you test instance, has an initial depth of 0".

Reply:
Thanks for the good suggestion @chu11. I slightly modified the example so that it shows the prompt after executing `flux start --test-size=1`, then added a note about `depth = 0` as suggested.